src/transforms/reduce/mod.rs

```diff
@@ -50,6 +50,9 @@ pub struct ReduceConfig {
     #[derivative(Default(value = "default_flush_period_ms()"))]
     pub flush_period_ms: Duration,
 
+    /// The maximum number of events to group together.
+    pub max_events: Option<usize>,
```

Comment:
This should probably be `Option<NonZeroUsize>`, as I can't think of a scenario where we'd want to set a limit to be 0, as that would cause the reducer to never flush if there was no `ends_when` condition set.

It can be `usize` elsewhere in the code, but we just want `NonZeroUsize` to catch erroneous values during deserialization.

Reply:
Good point! This made me realize that `max_events: 1` is a similar edge case. I can't think of a case for using it either (_maybe_ for effectively disabling the reduce without removing it, during development/troubleshooting?), but the current implementation would essentially ignore a 0 or 1 here and fall back to the start/end condition or expiration.

For 0 it makes sense to just disallow it as invalid config using `NonZeroUsize`. For `max_events: 1` I imagine it should behave the same as if `ends_when` always evaluates to true (i.e. immediately flush each event through).

Reply:
To your point, with `max_events: 1`, it will flush every event. While it's weird, it's at least flushing events, versus just effectively deadlocking if it's set to `max_events: 0`. Avoiding allowing users to deadlock themselves is really all I'm concerned about in terms of merging this in.. but we can always revisit behavior in the future since, as long as the field isn't specified _at all_, users don't need to use it.

Reply:
Ah yeah, I was totally spacing on the expiration setting being a backstop. 👍🏻